ccswbs/gus
# Remove Gatsby alias and use Drupal alias instead

## Summary of changes

### Frontend

- Remove alias-generating functions in `gatsby-node.js` (see e660028 for details)
- Remove mapping from `gatsby-config.js`
- Update `index.js` to use `node.path.alias` in `graphQL` query
- Replace tabs with spaces in updated files

### Backend

- Add composer patching capability as per https://duvien.com/blog/how-apply-patch-file-composer-based-drupal-89
- Update `composer.json` to add [paragraph revision fix patch](https://www.drupal.org/files/issues/2020-06-19/nested-paragraph-revision-jsonapi-workflow.patch)
- Update URL alias pattern for `Program` content type
- Add Gatsby Preview button to `Basic Page` and `Program` content types
- Add Linkit module and CHUG Linkit profile
- Update `Basic HTML` and `Full HTML` text formats to enable and use Linkit for internal links
- Update Gatsby settings to send `URL alias` to Gatsby Cloud as well

### Test plan

1. Go to multidev at https://gcloud-chug.pantheonsite.io/
2. Create a new basic page. The `Gatsby Preview` button should not be visible yet
3. Add a `general text` widget with text and save the new page as published
4. Go to the [gcloud site on Gatsby Cloud](https://www.gatsbyjs.com/dashboard/2e86c058-a370-4898-ae51-9698cd178e8d/sites/ac44984b-c63e-411d-8f1d-e96fb9305865/cmsPreview) and observe if a new Preview build occurred 
5. Go back to the page you created and update the existing `general text` widget
6. Save as draft
7. Click the Gatsby Preview button (which should now be visible)
8. The Preview page should open in a new window. Once the preview build completes, you should see your updated text
9. Repeat the above steps for a new program page
10. Repeat the above steps for a new basic page, but leave it in Draft mode throughout. It should still be previewable once saved and each successive save should trigger a new preview build that shows updated paragraph content.
